src/controller/python/chip/clusters/Attribute.py

```diff
@@ -47,9 +47,10 @@ class EventTimestampType(Enum):
 
 @unique
 class EventPriority(Enum):
-    DEBUG = 1
-    INFO = 2
-    CRITICAL = 3
+    DEBUG = 0
+    INFO = 1
+    CRITICAL = 2
+    INVALID = 3
```

Comment:
Where is INVALID used?

Reply:
This value is returned in the protocol, you can take a look
[](https://github.com/project-chip/connectedhomeip/actions/runs/4990492240/jobs/8949915611)

Reply:
in src/app/EventHeader.h defined
```
struct EventHeader
{
    ConcreteEventPath mPath;
    EventNumber mEventNumber     = 0;
    PriorityLevel mPriorityLevel = PriorityLevel::Invalid;
    Timestamp mTimestamp;
};
```

Reply:
Yes, but there it's used as just a "not initialized marker".  An actual event will never have that priority level.

Reply:
@bzbarsky-apple 
In the previous test, INVALID was not set, so it could not pass, and it is not ruled out that there is a BUG in other places.
https://github.com/project-chip/connectedhomeip/actions/runs/4990492240/jobs/8935776070

Reply:
in `src/app/ReadClient.cpp` , func `CHIP_ERROR ReadClient::ProcessEventReportIBs(TLV::TLVReader & aEventReportIBsReader)`

```
        else if (err == CHIP_END_OF_TLV)
        {
            EventStatusIB::Parser status;
            EventPathIB::Parser pathIB;
            StatusIB::Parser statusIBParser;
            ReturnErrorOnFailure(report.GetEventStatus(&status));
            ReturnErrorOnFailure(status.GetPath(&pathIB));
            ReturnErrorOnFailure(pathIB.GetEventPath(&header.mPath));
            ReturnErrorOnFailure(status.GetErrorStatus(&statusIBParser));
            ReturnErrorOnFailure(statusIBParser.DecodeStatusIB(statusIB));

            NoteReportingData();
            mpCallback.OnEventData(header, nullptr, &statusIB);
        }
```
`mPriorityLevel` is not set here

Reply:
> mPriorityLevel is not set here

Sure, since this is the "event does not actually have data; there's an error status instead" case.  In this case there is no priority available, or timestamp, or anything other than the path (which tells you which event you got an error for).  The ReadClient API is sort silly and uses an EventHeader to carry that path, but if there is no data the path is the _only_ part of the header which is non-garbage.  We should fix this API, perhaps,  but for the time being whatever glue sits on top of this API should not be propagating that garbage out: for errors it should pass on path+error, and for data it should pass on the event header + data.

Reply:
So looking at `OnEventData` in `src/controller/python/chip/clusters/attribute.cpp` the real problem is that it insists on calling a single callback for the "data" and "error" cases, and passes through the garbage data it has in the error case.  For comparison, the Darwin API's around OnEventData call different callbacks for the data and error cases, as does `src/controller/TypedReadCallback.h` and `src/controller/java/AndroidCallbacks.cpp`.